File: gin/library/captcha.go
package library

import (
	"fmt"
	"math/rand"
	"strings"
	"time"
)

type Captcha struct {
	txtChars string
}

/* 验证码 */
func (c *Captcha) Vcode() {
	c.txtChars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789"
	code := c.GetCode(4)
	fmt.Println(code, strings.ToLower(code))
}

/* 获取号码 */
func (c *Captcha) GetCode(num int) string {
	textNum := len(c.txtChars)
	code := ""
	r := rand.New(rand.NewSource(time.Now().UnixNano()))
	for i := 0; i < num; i++ {
		code = code + string(c.txtChars[r.Intn(textNum)])
	}
	return code
}

File: phalcon/library/Captcha.php
<?php
namespace Library;

use Service\Base;

class Captcha extends Base {

  static private $txtChars = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';

  /* 验证码 */
  static function Vcode() {
    $code = self::GetCode(4);
    self::Print($code, strtolower($code));
  }

  /* 获取号码 */
  static function GetCode(int $num): string {
    $code = '';
    for($i=0; $i<$num; $i++){
      $code .= substr(self::$txtChars,rand(0, strlen(self::$txtChars)-1), 1);
    }
    return $code;
  }

}